packages/components/utils/rename.js

```diff
@@ -5,9 +5,10 @@
 
 const { join } = require('path');
 const { readdirSync, renameSync } = require('fs');
+const escapeStringRegExp = require('escape-string-regexp');
 
 const [dir, search, replace] = process.argv.slice(2);
-const match = RegExp(search, 'g');
+const match = RegExp(escapeStringRegExp(search), 'g');
```

Comment:
Verified with my forked repository's codeQL analysis, the warnings for `Regular expression injection` were resolved. The `escapeStringRegExp` removes metacharacters such as ` /[|\\{}()[\]^$+*?.-]/g` from the argument

Reply:
what this file exactly does?
dont we need those characters anywhere?

Reply:
Its a seperate utility file used to rename large amounts of file in a folder, its mostly used for icons
<img width="741" alt="Screenshot 2023-05-17 at 11 53 04 AM" src="https://github.com/binary-com/deriv-app/assets/103016120/d5aad0cf-58d2-4717-98d7-583a04df2b1e">

Reply:
and you are escaping those characters from the new name, right?

Reply:
The metacharacters will be escaped within the `'string-to-search'` parameter, not the new name